### PR TITLE
Fix runtime configuration location on pipeline json

### LINKED ISSUE #1710: Update the pipeline schemas to remove redundant fields
**Describe the issue**
There are currently two places where a pipeline's runtime is configured. 
primary_pipeline-> app_data ->ui_data -> runtime 
primary_pipeline-> app_data -> runtime

We should merge the first into the latter to make things less ambiguous. We should also consider moving the runtime field completely out of the primary pipeline scope and up a level to make it clear that the runtime is associated with all pipelines(including supernodes) in the pipeline file.

This might get tricky when defining a generic pipeline as the runtime will be defined as null and the new validation service will only be able to perform a token validation (a check for components that ALL runtimes support) on the pipeline until submission time when a pipeline runtime is selected and a more thorough check can be done. At which point the latter runtime field would be overwritten with the selection(e.g. kfp, airflow, local). 
**Screenshots or log output**
```
      "app_data": {
        "ui_data": {
          "runtime": {
            "name": "kfp",
            "display_name": "Kubeflow Pipelines"
          },
          "comments": []
        },
        "version": 3,
        "name": "untitled4",
        "runtime": "kfp",
        "runtime-config": "kfp",
        "source": "invalid_node_property_image_name.pipeline"
      },
      "runtime_ref": ""
    }
```

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: master branch
 - Operating system: macos

### LINKED ISSUE #1772: Runtime specific pipeline save runtime information in a different place then expected
**Describe the issue**
Runtime specific pipeline save runtime information in `primary_pipeline["app_data"]["ui_data"]["runtime"]["name"]` instead of the expected location `primary_pipeline["app_data"]["runtime"]`.

Invalid runtime specific pipeline:
image

Properly submitted pipeline: 
image

### PR DESCRIPTION
**Pull request by lresende**
Fixes #1772 
Fixes #1710 

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Review by kevin-bates — COMMENTED** [review]
The changes look good.  Are the deleted files ones that could have been deleted a while back when things were moved to the server (and unrelated to this issue)?

This change seems to warrant a version bump and auto-migration - is that correct?  Or will the back-end be updated to tolerate getting the runtime name from the `app_data.ui_data.runtime.name` if not present in `app_data` directly?  Either way, it seems like more changes are necessary if pipelines using `app_data.ui_data.runtime.name` exist in the wild.

A server-side test seems useful here if I'm understanding the flow.

**Comment by lresende** [comment]
> The changes look good. Are the deleted files ones that could have been deleted a while back when things were moved to the server (and unrelated to this issue)?

Correct, I can split this into a different pr if needed.

> 
> This change seems to warrant a version bump and auto-migration - is that correct? Or will the back-end be updated to tolerate getting the runtime name from the `app_data.ui_data.runtime.name` if not present in `app_data` directly? Either way, it seems like more changes are necessary if pipelines using `app_data.ui_data.runtime.name` exist in the wild.
> 

This changed is not on a release. This is a fix required to maintain backward compatibility.

**Comment by lresende** [comment]
> > The changes look good. Are the deleted files ones that could have been deleted a while back when things were moved to the server (and unrelated to this issue)?
> 
> Correct, I can split this into a different pr if needed.

Moved the cleanup to another PR. Also found another obsolete file.

**Comment by marthacryan** [comment]
Didn't realize you'd fixed the CI issue - just forced pushed to get rid of my fix 😄

File: packages/pipeline-editor/src/index.ts [hunk 1]
```diff
@@ -200,12 +200,12 @@ const extension: JupyterFrontEndPlugin<void> = {
                       comments: [],
                       runtime: args.runtime
```
Comment (marthacryan) [inline packages/pipeline-editor/src/index.ts:201]: Couldn't this entire field be removed?

**Review by marthacryan — APPROVED** [review]
Tested locally and it worked, code LGTM

**Review by kevin-bates — APPROVED** [review]

**Pull request merged**